### PR TITLE
add the jmx-monitoring plugin

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

**Comment by matriv** [comment]
I would add caching to prevent the SQL execution a certain period of time (eg. 100-200) to prevent badly configured monitoring tools from "attacking" the jmx MBean but maybe is over-engineering.
@mfussenegger what do you think?

**Comment by kovrus** [comment]
I think we do not need the caching here. If we do it for such insignificant period amount of type, it won't be helpful. If we increase the caching time, we won't get actual metric values at the time of query.

File: jmx-monitoring/NOTICE [hunk 1]
```diff
@@ -0,0 +1,33 @@
+CrateDB Sample Enterprise Module
```
Comment (mfussenegger) [inline jmx-monitoring/NOTICE]: Sample Enterprise Module?

File: jmx-monitoring/src/main/java/io/crate/plugin/CrateMonitoringPlugin.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public Collection<Module> nodeModules() {
+        if (isEnterprise()) {
+            return ImmutableList.of(new CrateMonitoringModule());
+        }
+        return Collections.emptyList();
+    }
+
+    private boolean isEnterprise() {
+        return settings.getAsBoolean("license.enterprise", false);
```
Comment (mfussenegger) [inline jmx-monitoring/src/main/java/io/crate/plugin/CrateMonitoringPlugin.java]: Result could be assigned to a variable in the constructor

**Comment by kovrus** [comment]
@mfussenegger yes, a small one. The RowReceiver interface has changed since then, so it was adapted.

**Pull request merged**